src/python/pants/init/logging.py

```diff
@@ -37,16 +38,32 @@ def flush(self) -> None:
 
 
 class _ExceptionFormatter(Formatter):
-    """Uses the `--print-stacktrace` option to decide whether to render stacktraces."""
+    """Possibly render the stacktrace and possibly give debug hints, based on global options."""
 
-    def __init__(self, print_stacktrace: bool):
+    def __init__(self, level: LogLevel, *, print_stacktrace: bool, local_cleanup: bool) -> None:
         super().__init__(None)
+        self.level = level
         self.print_stacktrace = print_stacktrace
+        self.local_cleanup = local_cleanup
 
     def formatException(self, exc_info):
-        if self.print_stacktrace:
-            return super().formatException(exc_info)
-        return "\n(Use --print-stacktrace to see more error details.)"
+        stacktrace = super().formatException(exc_info) if self.print_stacktrace else ""
+
+        debug_instructions = []
+        if not self.print_stacktrace:
+            debug_instructions.append("--print-stacktrace for more error details")
+        if self.local_cleanup:
+            debug_instructions.append("--no-process-execution-local-cleanup to inspect chroots")
+        if self.level not in {LogLevel.DEBUG, LogLevel.TRACE}:
+            debug_instructions.append("-ldebug for more logs")
+        debug_instructions = (
+            f"Use {' and/or '.join(debug_instructions)}. " if debug_instructions else ""
+        )
+
+        return (
+            f"{stacktrace}\n\n({debug_instructions}See {unbracketed_docs_url('troubleshooting')} for common issues. "
+            f"Consider reaching out for help: {unbracketed_docs_url('getting-help')}.)"
```

Comment:
One of the cases the brackets are meant to defend against is the trailing period here being interpreted as part of the URL... although most URL parsers probably do the right thing with that at this point. It probably continues to be a good default behavior.

Reply:
True. FWIT, iterm2 does the right thing, but not sure about other shells. I can go back to `bracketed_docs_url` if we want - I only didn't like how much nesting of brackets there was.

Reply:
Hm, so looking at how this triggers in practice, the `-ldebug` and `--print-stacktrace` messages are universally helpful, but the `--no-process-execution-local-cleanup` bit triggers in a bunch of cases where it won't be useful: for example, when you get a flag name wrong.
```
pants.option.errors.UnknownFlagsError: Unknown flags --blah on global scope

(Use --no-process-execution-local-cleanup to inspect chroots and/or -ldebug for more logs. See https://www.pantsbuild.org/v2.6/docs/troubleshooting for common issues. Consider reaching out for help: https://www.pantsbuild.org/v2.6/docs/getting-help.)
```

Reply:
What about rewording "to inspect chroots if Pants ran any subprocesses"?

Reply:
Eh, sure. But that's noisy unless we know that it ran a process locally.

If possible, enriching errors here: https://github.com/pantsbuild/pants/blob/0429258b181986eed856ae45af93b776727774a0/src/rust/engine/process_execution/src/local.rs#L239-L284 ... would make the most sense to me (although we would not be able to fail for a non-zero exit code there)... or perhaps here: https://github.com/pantsbuild/pants/blob/0429258b181986eed856ae45af93b776727774a0/src/python/pants/engine/process.py#L241-L260

Although inspecting successful processes might make sense in some cases (because the process succeeded, but you got invalid data), I think that it is primarily failing processes for which it is most obviously useful to render that help. In the case of succeeding processes, it's not obvious _which_ sandbox to look at: backtracking the invalid-but-successful data is a very advantage usecase.